streaming/src/main/scala/org/apache/spark/streaming/scheduler/ReceiverTracker.scala

```diff
@@ -295,6 +296,14 @@ class ReceiverTracker(ssc: StreamingContext, skipReceiverLaunch: Boolean = false
         val receiver = iterator.next()
         val supervisor = new ReceiverSupervisorImpl(
           receiver, SparkEnv.get, serializableHadoopConf.value, checkpointDirOption)
+
+        // Due to SPARK-5205, `Receiver` stage can not be stopped properly, we need to add a
```

Comment:
It's fine to be a bit more explicit / verbose about what's going on here, so I'd maybe update this comment to say something like this:

> To ensure that receivers are stopped properly when their tasks / stages are killed, register a callback to stop the receiver when its task is interrupted.  See SPARK-5205 for more details.